src/SmartComponents/FormComponents/BulkSelectButton.js

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+import { Button } from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import useFormApi from '@data-driven-forms/react-form-renderer/use-form-api';
+import useFieldApi from '@data-driven-forms/react-form-renderer/use-field-api';
+import { SELECT_ALL } from '../../Utilities/constants';
+import './BulkSelectButton.scss';
+
+// eslint-disable-next-line no-unused-vars
+const BulkSelectButton = (props) => {
+  const {
+    group,
+    section,
+    label,
+    input: { onChange, checked, ...input },
+  } = useFieldApi({
+    ...props,
+    type: 'button',
+  });
+  const formOptions = useFormApi();
+  const [buttonName, setButtonName] = useState(label);
+  useEffect(() => {
+    let isSubscribed = false;
+    formOptions.batch(() => {
+      formOptions.getRegisteredFields().forEach((field) => {
+        if (
+          !field.includes(SELECT_ALL) &&
+          ((field.includes(group) && field.includes(section)) ||
+            (field === 'is_subscribed' && // a temporary condition for RHEL Advisor email pref.
+              group === 'rhel' &&
+              section == 'advisor'))
+        ) {
+          if (!formOptions.getFieldState(field).value) {
+            isSubscribed = true;
+          }
+        }
+      });
+    });
+    setButtonName(isSubscribed == true ? 'Select all' : 'Deselect all');
```

Comment:
I would introduce constants for "Select all" and "Deselect all" so you can use them across the file.